### PR TITLE
Delete objects in different namespaces

### LINKED ISSUE #564: Resources are not removed with a namespace specified in the yaml when upgrading
From https://github.com/replicatedhq/kotsadm/issues/1408

Steps to reproduce:
1. create a release, specify a namespace other than the target namespace in the yaml for a deployment
2. install that release
3. create a new release and remove that deployment
4. install the new release, the deployment is not removed

additionalNamespaces does not resolve the issue

diffAndRemovePreviousManifests does not call docsByNamespace
https://github.com/replicatedhq/kotsadm/blob/master/operator/pkg/client/deploy.go#L105

### PR DESCRIPTION
**Pull request by emosbaugh**
Fixes #564

### CODE REVIEW

File: kotsadm/operator/pkg/client/deploy.go [hunk 1]
```diff
@@ -44,18 +47,23 @@ func (c *Client) diffAndRemovePreviousManifests(applicationManifests Application
 		return errors.Wrap(err, "failed to decode manifests")
 	}
 
+	targetNamespace := c.TargetNamespace
+	if applicationManifests.Namespace != "." {
```
Comment (laverya) [inline kotsadm/operator/pkg/client/deploy.go:51]: != `"."`, or != `""`?

Comment (laverya) [reply kotsadm/operator/pkg/client/deploy.go:51]: I see that that's what it was before, but it feels odd

Comment (emosbaugh) [reply kotsadm/operator/pkg/client/deploy.go:51]: ya, idk. but its correct

File: kotsadm/operator/pkg/client/gvkn.go [hunk 2]
```diff
@@ -20,20 +19,24 @@ type OverlySimpleGVKWithName struct {
 }
 
 type OverlySimpleMetadata struct {
-	Name      string `yaml:"name"`
-	Namespace string `yaml:"namespace"`
+	Name        string                 `yaml:"name"`
+	Namespace   string                 `yaml:"namespace"`
+	Annotations map[string]interface{} `json:"annotations"`
```
Comment (laverya) [inline kotsadm/operator/pkg/client/gvkn.go:24]: why'd this change? It doesn't seem to be used anywhere in this PR

**Pull request merged**